hypothesis-python/src/hypothesis/internal/conjecture/engine.py

```diff
@@ -1974,6 +1975,33 @@ def try_delete_range(k):
             # next example to be undeletable.
             i += 1
 
+    @shrink_pass
+    def block_deletion(self):
+        """A secondary deletion pass which attempts to delete short contiguous
+        ranges of blocks. This is able to violate boundaries in a way that
+        adaptive_example_deletion cannot.
+
+        An example of where this is useful is that if we are generating
+        a list of lists, we can merge consecutive elements, e.g.
+        replacing [[a], [b]] with [[a, b]], because this pass will
+        simultaneously delete the last block of the [a] and the first
+        block of the [b], effectively joining the two lists together.
+        """
+        i = 0
+        while i < len(self.shrink_target.blocks):
+            j = min(i + 4, len(self.shrink_target.blocks) - 2)
+            while j >= i:
+                u, v = self.shrink_target.blocks[i]
+                r, s = self.shrink_target.blocks[j]
```

Comment:
Why do you need `u`, `v`, `r`, `s` if none of those are used?

Reply:
If you want to ensure that whose elements are two-items iterable, may be explicitly check for their length? Better error message would help in debug if it eventually occurs.

Reply:
Whoops, because that code is entirely wrong but just happened to work for fixing the test! I've now fixed it.

Those values are always of length 2. Blocks are explicitly pairs of endpoints.